lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -595,4 +598,78 @@
         }
         return o;
     });
+
+    platform.registerFunction(platform.FunctionKind.Math_min, function (value1, value2, ...values) {
+        // #sec-math.min
+
+        // If no arguments are given, the result is positive infinity
+        // If any value is NaN, the result is NaN. 
+        // The comparison of values to determine the smallest value is done using the Abstract Relational Comparison algorithm except that +0 is considered to be larger than -0.
+        if (arguments.length === 0 ) {
+            return __chakraLibrary.positiveInfinity;
+        }
+
+        if (value1 === null) value1 = 0;
+        if (arguments.length === 1) {
+            if (__chakraLibrary.isNaN(value1)) return NaN;
+            else return value1;
```

Comment:
We still need to coerce some things to numbers before returning them. For example Math.min(true) is 1.

Reply:
Strings also if they're parseable. I think the cleanest option would be to expose a function __chakraLibrary.toNumber that does #sec-tonumber and call it all the places that you're currently doing null checks.

---
In reply to: [222838184](https://github.com/Microsoft/ChakraCore/pull/5731#discussion_r222838184) [](ancestors = 222838184)

Reply:
That is interesting because in the ECMAScript, it says "If any value is NaN, the result is NaN."

Reply:
You can coerce to number by doing numberOfValue = +value
e.g.
print (+"5");
print (+"a");

Prints 5 and NaN, whilst Number.isNaN("a") is false Number.isNaN(+"a") is true.

(Note after coercing to number with + you can then use  value !== value as the NaN check if you want to avoid the extra function calls)

Reply:
Good point @rhuanjl , I forgot the language already has a simple syntactic way to convert to numbers. Thanks!

Reply:
1a82165a5e48c87f6559f5c109d3810c62b0800e